lib/tests.js

```diff
@@ -0,0 +1,405 @@
+                console.log(`Second listener accessed message ${data.message}`);
+
+                if (data.message === "multiMsg") {
+                    dealWithData(data);
+                    instanceTwo.removeAllListeners();
+                    instanceTwo.stopListening();
+                }
+            });
+            instance.startListening();
+            instanceTwo.startListening();
+        });
+    }
+
+
+    /**
+     * Uses multiple listeners to ensure both listeners do not get the same
+     * message.
+     *
+     * @param {Function} callback
+     */
+    function multipleListenerTest(callback) {
+        instance.on("error", (err) => {
+            instance.removeAllListeners();
+            callback(err);
+        });
+        instance.connect((err) => {
+            if (err) {
+                instance.removeAllListeners();
+                callback(err);
+
+                return;
+            }
+
+            console.log("Sending test message for multiple listener test.");
+            instance.sendMessage("multiMsg", config.topics, (testErr) => {
+                if (testErr) {
+                    instance.removeAllListeners();
+                    callback(testErr);
+
+                    return;
+                }
+
+                multipleListeners((multiErr) => {
+                    instance.removeAllListeners();
+                    instance.disconnect();
+                    callback(multiErr);
+                });
+            });
+        });
+    }
+
+
+    /**
+     * This tests the deduplication functionality. It ensures if a duplicate
+     * message is sent, it is not inserted and requeued is set to true.
+     * It also tests the requeued process. The "requeued" column exists because
+     * we want to run a job one more time but only if the message is being
+     * processed.
+     *
+     * @param {Function} callback
+     */
+    function deduplicationTest(callback) {
+        var emitted;
+
+        emitted = false;
+        instance.on("error", (err) => {
+            instance.removeAllListeners();
+            callback(err);
+        });
+        instance.connect((err) => {
+            if (err) {
+                instance.removeAllListeners();
+                callback(err);
+
+                return;
+            }
+
+            console.log("Sending test message.");
+            instance.sendMessage("message", config.topics, (testErr) => {
+                if (testErr) {
+                    callback(testErr);
+
+                    return;
+                }
+
+                console.log("Test message inserted successfully");
+            });
+            console.log("Testing duplicate message");
+            instance.on("data", (data) => {
+                instance.sendMessage("message", config.topics, (fail) => {
+                    if (fail && !fail.message === "Could not send message.") {
+                        callback(new Error("Was able to insert a duplicate message!"));
```

Comment:
I don't get this message. It wouldn't have inserted a duplicate message in this scenario would it?